### PR TITLE
Added "Add Mature/Remove Mature" to the mod menu

### PR DESCRIPTION
**Pull request by Cleover**
Added the ability to add and remove the mature role from a user in the mod menu list, similar to the verify option

### CODE REVIEW

File: configs/snowflakeMapType.d.ts [hunk 1]
```diff
@@ -50,4 +50,9 @@ export default interface snowflakeMap {
    * verified by mods)
    */
   Verified_Roles: string[];
+
+  /**
+   * A list of roles that members verified with /mod user Add Mature / Remove Mature should be Given / Removed
```
Comment (elise32) [inline configs/snowflakeMapType.d.ts:55]: 'A list of roles that members [use a different word than verified to avoid confusion with the verification system] with /mod user add_mature/remove_mature should be given/removed'

File: src/resources/commandTemplates/templates.ActionCommand.ts [hunk 2]
```diff
@@ -325,7 +325,19 @@ export default class ActionCommand extends ResponsiveSlashCommandSubcommandBuild
       (member: GuildMember) => Promise<boolean>,
       (member: GuildMember, reason: string, days?: number) => Promise<boolean>,
       ExtraActionOptions
-    ]
+    ],
+    [
+      APIApplicationCommandOptionChoice<string>,
+      (member: GuildMember) => Promise<boolean>,
+      (member: GuildMember, reason: string) => Promise<boolean>,
```
Comment (elise32) [inline src/resources/commandTemplates/templates.ActionCommand.ts:332]: not sure if reason here and on line 338 are necessary

File: src/resources/commandTemplates/templates.ActionCommand.ts [hunk 3]
```diff
@@ -399,6 +411,36 @@ export default class ActionCommand extends ResponsiveSlashCommandSubcommandBuild
...
+        },
+        async (member) => {
+          return member.manageable;
+        },
+        async (member, reason) => {
+          if (!member.manageable) return false;
+          const SNOWFLAKE_MAP = await getSnowflakeMap();
+          return !!(await member.roles.add(SNOWFLAKE_MAP.Mature_Roles, reason));
+        },
+        { emoji: ':white_check_mark:', pastTense: 'added to the mature role' },
```
Comment (elise32) [inline src/resources/commandTemplates/templates.ActionCommand.ts:427]: not sure where 'added to the mature role', 'removed from the mature role' are used but i'd rephrase the grammar

File: src/resources/commandTemplates/templates.ActionCommand.ts (hunk 2 above)
Comment (Cleover) [reply src/resources/commandTemplates/templates.ActionCommand.ts:332]: its used in /verify as well, unsure why but ill keep it in to keep up with the code skyler made

File: configs/snowflakeMapType.d.ts (hunk 1 above)
Comment (Cleover) [reply configs/snowflakeMapType.d.ts]: Fixed!

File: src/resources/commandTemplates/templates.ActionCommand.ts (hunk 3 above)
Comment (Cleover) [reply src/resources/commandTemplates/templates.ActionCommand.ts]: Resolved!

**Review by elise32 — APPROVED** [review]
